### PR TITLE
Catch and log exceptions from agent loading

### LINKED ISSUE #403: Be more resilient to Http2Agent installation failures
Saw this stacktrace when running in docker container `alpine-java:8u121-t95` with the Oracle JDK:

```
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
    at com.ea.agentloader.AgentLoader$1.loadAgent(AgentLoader.java:244)
    at com.ea.agentloader.AgentLoader.loadAgent(AgentLoader.java:77)
    at com.ea.agentloader.AgentLoader.loadAgentClass(AgentLoader.java:148)
    at com.ea.agentloader.AgentLoader.loadAgentClass(AgentLoader.java:102)
    at com.palantir.remoting2.http2.Http2Agent.install(Http2Agent.java:26)
    at com.palantir.remoting2.jaxrs.FeignJaxRsClientBuilder.<clinit>(FeignJaxRsClientBuilder.java:65)
    ... 48 more
Caused by: java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at com.ea.agentloader.AgentLoader$1.loadAgent(AgentLoader.java:240)
    ... 53 more
Caused by: java.lang.RuntimeException: Can't attach to this jvm. Add -javaagent:/tmp/javaagent.com.palantir.remoting2.http2.Http2Agent9090971280470105146.jar to the commandline
    at com.ea.agentloader.AgentLoaderHotSpot.loadAgent(AgentLoaderHotSpot.java:74)
    ... 58 more
Caused by: com.sun.tools.attach.AgentInitializationException: Agent JAR loaded but agent failed to initialize
    at sun.tools.attach.HotSpotVirtualMachine.loadAgent(HotSpotVirtualMachine.java:121)
    at com.ea.agentloader.AgentLoaderHotSpot.loadAgent(AgentLoaderHotSpot.java:65)
    ... 58 more
```

and saw this exception when running in docker container `openjdk:8-alpine` with the OpenJDK:

```
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
    at com.ea.agentloader.AgentLoader$1.loadAgent(AgentLoader.java:244)
    at com.ea.agentloader.AgentLoader.loadAgent(AgentLoader.java:77)
    at com.ea.agentloader.AgentLoader.loadAgentClass(AgentLoader.java:148)
    at com.ea.agentloader.AgentLoader.loadAgentClass(AgentLoader.java:102)
    at com.palantir.remoting2.http2.Http2Agent.install(Http2Agent.java:26)
    at com.palantir.remoting2.jaxrs.FeignJaxRsClientBuilder.<clinit>(FeignJaxRsClientBuilder.java:65)
    ... 48 more
Caused by: java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at com.ea.agentloader.AgentLoader$1.loadAgent(AgentLoader.java:240)
    ... 53 more
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
    at com.ea.agentloader.AgentLoaderHotSpot.getVirtualMachine(AgentLoaderHotSpot.java:113)
    at com.ea.agentloader.AgentLoaderHotSpot.loadAgent(AgentLoaderHotSpot.java:56)
    ... 58 more
Caused by: java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
    at com.ea.agentloader.AgentLoaderHotSpot.getVirtualMachine(AgentLoaderHotSpot.java:104)
    ... 59 more
Caused by: com.sun.tools.attach.AttachNotSupportedException: Unable to get pid of LinuxThreads manager thread
    at sun.tools.attach.LinuxVirtualMachine.<init>(LinuxVirtualMachine.java:86)
    ... 64 more
```

To fix I would propose:
1) for the short term, put a try/catch around https://github.com/palantir/http-remoting/blob/develop/http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java#L28 to be more resilient to installation failures
2) more deeply investigate the failures and attempt to fix in whichever upstream project that has this bug (ea-agent-loader, JDK, jetty-alpn-agent)

### PR DESCRIPTION
**Pull request by ash211**
Fixes #403

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/http-remoting/404)

### CODE REVIEW

**Review by uschi2000 — APPROVED** [review]

File: http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java [hunk 1]
```diff
@@ -25,8 +29,12 @@ public static synchronized void install() {
             return;
         }
 
-        AgentLoader.loadAgentClass(Http2Agent.class.getName(), "");
-        hasBeenInstalled = true;
+        try {
+            AgentLoader.loadAgentClass(Http2Agent.class.getName(), "");
+            hasBeenInstalled = true;
+        } catch (Exception e) {
+            logger.error("Unable to dynamically install jetty-alpn-agent via ea-agent-loader, proceeding anyway...", e);
```
Comment (markelliot) [inline http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java]: warn

File: http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java [hunk 2]
```diff
@@ -7,11 +7,15 @@
...
 import java.lang.instrument.Instrumentation;
 import org.mortbay.jetty.alpn.agent.Premain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** A very simple wrapper around the jetty-alpn-agent for dynamic loading. */
 public final class Http2Agent {
     private Http2Agent() {}
 
+    private static final Logger logger = LoggerFactory.getLogger(Http2Agent.class);
```
Comment (markelliot) [inline http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java]: logger -> log

Comment (ash211) [reply http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java]: done -- should this opinion be in gradle-baseline?

File: http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java (hunk 1 above)
Comment (ash211) [reply http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java]: done

**Review by ash211 — COMMENTED** [review]
Updated to address comments

File: http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java (hunk 2 above)
Comment (markelliot) [reply http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java]: yes, I've been thinking about writing an errorprone check but haven't gotten around to it.

Comment (ash211) [reply http2-agent/src/main/java/com/palantir/remoting2/http2/Http2Agent.java]: https://github.com/palantir/gradle-baseline/issues/159

**Pull request merged**